[docs][notifications] Add Android and iOS notification object examples (#31211)
expo/expo: docs/pages/push-notifications/receiving-notifications.mdx

description: Learn how to respond to a notification received by your app and take action based on the event.
---

> import { Collapsible } from '~/ui/components/Collapsible';

Reviewer: ```suggestion
import { Collapsible } from '~/ui/components/Collapsible';
import { CODE } from '~/ui/components/Text';
```

}, []);
```

> <Collapsible summary="Android Notification Object Example from addNotificationReceivedListener">

Reviewer: ```suggestion
<Collapsible summary={<>Android notification object example from <CODE>addNotificationReceivedListener</CODE></>}>
```

Sentence case is preferred over title case for section headings as per our writing style guide: https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#headings

> Here's a sample of the notification object received by the callback function when using `Notifications.addNotificationReceivedListener`:

Reviewer: ```suggestion
Sample of the `notification` object received by the callback function when using `Notifications.addNotificationReceivedListener`:
```

We prefer using contractions with words like "you" or "we". For rest, we should avoid them. Also, the sample snippet below is using a mix of different code languages with JSX marked as primary code language (which is not accurate in this case. Let's remove the console log statement and instead use code syntax highlight and use comments in the JSON snippets.

> ```jsx

Reviewer: ````suggestion
```json
````

> console.log(notification);

Reviewer: ```suggestion
// console.log(notification);
```

          "scopeKey": "@betoatexpo/expo-notifications-app",
          "experienceId": "@betoatexpo/expo-notifications-app",
          "projectId": "51092087-87a4-4b12-8008-145625477434"
        },
        "to": null,
        "ttl": 0,
        "collapseKey": "dev.expo.notificationsapp",
        "messageType": null,
        "priority": 2,
        "from": "115310547649",
        "messageId": "0:1724782348220771%0f02879c0f02879c"
      },
      "channelId": "default",
      "type": "push"
    },
    "content": {
      "autoDismiss": true,
      "title": "Chat App",
      "badge": null,
      "sticky": false,
      "sound": "default",
      "body": "New message from John Doe",
      "subtitle": null,
      "data": {
        "senderId": "user123",
        "senderName": "John Doe",
        "messageId": "msg789",
        "conversationId": "conversation-456",
        "messageType": "text",
        "timestamp": 1724766427
      }
    },
    "identifier": "0:1724782348220771%0f02879c0f02879c"
  },
  "date": 1724782348210

> }

Reviewer: ````suggestion
}
```
````

Let's break this snippet since the next snippet is not a direct output of this one.

> // You can access the notification custom data as follows:

Reviewer: ````suggestion
You can directly access the notification custom data by logging the `notification.request.content.data` object:

```json
````

It's always best to describe a code snippet in words so that people can read it or it grabs their attention. Often times people prefer to skim and important info can be easily missed out by them.

> console.log(notification.request.content.data);

Reviewer: ```suggestion
// console.log(notification.request.content.data);
```

{
  "senderId": "user123",
  "senderName": "John Doe",
  "messageId": "msg789",
  "conversationId": "conversation-456",
  "messageType": "text",
  "timestamp": 1724766427
}
```

</Collapsible>

> <Collapsible summary="iOS Notification Object Example from addNotificationReceivedListener">

Reviewer: ```suggestion
<Collapsible summary={<>iOS notification object example from <CODE>addNotificationReceivedListener</CODE></>}>
```

> Here's a sample of the notification object received by the callback function when using `Notifications.addNotificationReceivedListener`:

Reviewer: ```suggestion
Sample of the `notification` object received by the callback function when using `Notifications.addNotificationReceivedListener`:
```

> ```jsx
> console.log(notification);

Reviewer: ````suggestion
```json
// console.log(notification);
````

            "title": "Chat App",
            "launch-image": "",
            "body": "New message from John Doe"
          },
          "sound": "default"
        },
        "body": {
          "messageId": "msg789",
          "timestamp": 1724766427,
          "messageType": "text",
          "senderId": "user123",
          "senderName": "John Doe",
          "conversationId": "conversation-456"
        }
      }
    },
    "identifier": "3AEB849E-9059-4D09-BC3B-9A0B104CF062",
    "content": {
      "body": "New message from John Doe",
      "sound": "default",
      "launchImageName": "",
      "badge": 1,
      "subtitle": "Hey there! How's your day going?",
      "title": "Chat App",
      "data": {
        "conversationId": "conversation-456",
        "senderName": "John Doe",
        "senderId": "user123",
        "messageType": "text",
        "timestamp": 1724766427,
        "messageId": "msg789"
      },
      "summaryArgument": null,
      "categoryIdentifier": "",
      "attachments": [],
      "interruptionLevel": "active",
      "threadIdentifier": "",
      "targetContentIdentifier": null,
      "summaryArgumentCount": 0
    }
  },
  "date": 1724798493.0589335

> }

Reviewer: ````suggestion
}
```
````

> // You can access the notification custom data as follows:
> console.log(notification.request.content.data);

Reviewer: ````suggestion
You can directly access the notification custom data by logging the `notification.request.content.data` object:

```json
// console.log(notification.request.content.data);
````
